S3/S3.py

```diff
@@ -1848,6 +1848,11 @@ def send_file(self, request, stream, labels, buffer = '', throttle = 0,
             sha256_hash = checksum_sha256_file(stream, offset, size_total)
         request.body = sha256_hash
 
+        # Provide the checksum with the request. This is important for buckets that have 
+        # Object Lock enabled.
+
+        headers['x-amz-checksum-sha256'] = sha256_hash_to_base64(sha256_hash)
```

Comment:
As someone could use s3cmd as library this line can override the header potentially comes with request argument's headers attribute. I think it's better to check if it's currently has values:
```python
if not headers.get("x-amz-checksum-sha256"):
    headers['x-amz-checksum-sha256'] = sha256_hash_to_base64(sha256_hash)
```